app/celer-sim/celer-sim.cc

```diff
@@ -131,16 +134,23 @@ void run(std::istream* is, std::shared_ptr<OutputRegistry> output)
             activate_device_local();
 
             // Run a single event on a single thread
-            CELER_TRY_HANDLE(result.events[event] = run_stream(
-                                 StreamId(get_openmp_thread()), EventId(event)),
-                             capture_exception);
+            CELER_TRY_HANDLE(
+                (run_input->transporter_result)
+                    ? result.events[event] = run_stream(
+                          StreamId(get_openmp_thread()), EventId(event))
+                    : run_stream(StreamId(get_openmp_thread()), EventId(event)),
+                capture_exception);
```

Comment:
```suggestion
            TransporterResult event_result;
            CELER_TRY_HANDLE(event_result = run_stream(
                          StreamId(get_openmp_thread()), EventId(event)),
                capture_exception);
            if (run_input->transporter_result)
            {
                result.events[event] = event_result;
            }
```

Reply:
Do you think this copy could be slow if the event is large? I tried to avoid a copy.

Reply:
You're right, I should have std::move. The readability trumps potential (small) copy cost and increased object size